src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -134,7 +175,34 @@ public void Dispose()
             }
         }
 
-        private async Task<string> GetLastPollStatusAsync()
+        private async Task MoveToPoisonQueueAsync(ApiHubFile apiHubFile)
+        {
+            CloudQueueClient queueClient = CloudStorageAccount.Parse(this._config.StorageConnectionString).CreateCloudQueueClient();
+
+            var queue = queueClient.GetQueueReference(PoisonQueueName);
+
+            await queue.CreateIfNotExistsAsync();
+
+            var fileStatus = new ApiHubFileInfo
+            {
+                FilePath = apiHubFile.Path,
+                FunctionName = this._functionName,
+                Connection = this._connectionStringSetting
```

Comment:
It's not the actual connection string. It's the connectionStringSetting aka the key.